src/main/java/net/apnic/whowas/App.java

```diff
@@ -112,15 +98,19 @@ public TaskExecutor taskScheduler() {
     ApplicationContext context;
 
     @Autowired
-    private JdbcOperations jdbcOperations;
-
     private RipeDbLoader dbLoader;
+
     private Exception lastDbException = null;
 
     @PostConstruct
-    public void initialise() {
-        dbLoader = new RipeDbLoader(jdbcOperations, -1L);
-        executorService.execute(this::buildTree);
+    public void initialise() throws ExecutionException, InterruptedException {
+        /*
+          Scoping the initial load of data to @PostConstruct
+          results in the loading completing before the servlet context is started,
+          preventing requests from being served from partially loaded data.
+        */
+        Future initialDataLoaded = executorService.submit(this::buildTree);
```

Comment:
Nice. But missing the dbLoader assignment?

Reply:
Where should I be looking for the bean create though? The \@Component constructor needs a JdbcOperations, but it's not \@Autowired.

Reply:
That is a new spring feature. @Component bean constructor parameters can be omitted now. 

https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-spring-beans-and-dependency-injection.html